cmd/mistryd/server.go

```diff
@@ -494,30 +494,51 @@ func PruneZombieBuilds(cfg *Config) error {
 	return nil
 }
 
-func getProjects(path string) ([]string, error) {
-	folders, err := ioutil.ReadDir(path)
+func getProjects(cfg *Config) ([]string, error) {
+	root := cfg.ProjectsPath
+	folders, err := ioutil.ReadDir(root)
 	if err != nil {
 		return nil, err
 	}
+
 	projects := []string{}
+
 	for _, f := range folders {
-		if f.IsDir() {
-			projects = append(projects, f.Name())
+		if !f.IsDir() {
```

Comment:
The logic in this loop could be extracted to a separate method, sth like `isValidProject`, and could serve as a "definition" of a valid project.

Reply:
I think this would make more sense if it was used in other places too. As it is now, I think the added level of indirection doesn't worth it.